subprojects/build-init/src/main/java/org/gradle/buildinit/plugins/internal/BuildScriptBuilder.java

```diff
@@ -130,10 +137,29 @@ public void generate() {
                         writer.println(" */");
 
                         // Plugins
-                        for (Map.Entry<String, String> entry : plugins.entrySet()) {
+                        if (!plugins.isEmpty()) {
                             writer.println();
-                            writer.println("// " + entry.getValue());
-                            writer.println("apply plugin: '" + entry.getKey() + "'");
+                            writer.println("plugins {");
+                            for (Iterator<Map.Entry<String, String>> it = plugins.entrySet().iterator(); it.hasNext();) {
+                                Map.Entry<String, String> entry = it.next();
+                                writer.println("    // " + entry.getValue());
+                                switch (dsl) {
+                                    case KOTLIN:
+                                        if (entry.getKey().matches("[a-z]+")) {
+                                            writer.println("    " + entry.getKey());
+                                        } else {
+                                            writer.println("    `" + entry.getKey() + "`");
+                                        }
+                                        break;
+                                    case GROOVY:
+                                    default:
```

Comment:
⭕️ What's the `default` case if not `KOTLIN` nor `GROOVY`?

Reply:
Could be `null`.
But we could guard against that and throw if not `KOTLIN` nor `GROOVY`.

Reply:
If this was kotlin you wouldn't need the default because of the exhaustive search the compiler does.